desktop/ui/src/main/java/org/datacleaner/windows/FixedWidthDatastoreDialog.java

```diff
@@ -197,8 +189,60 @@ private void onSettingsUpdated(boolean autoDetectEncoding) {
 	}
 
 	@Override
+	protected byte[] getSampleBuffer() {
+		final File file = new File(getFilename());
+		final int bufferSize = getBufferSize();
+		byte[] bytes = new byte[bufferSize];
+		FileInputStream fileInputStream = null;
+
+		try {
+			fileInputStream = new FileInputStream(file);
+			int startPosition = getStartPosition();
+			fileInputStream.skip(startPosition);
+			int bytesRead = fileInputStream.read(bytes, 0, bufferSize);
+
+			if (bytesRead != -1 && bytesRead <= bufferSize) {
+				bytes = Arrays.copyOf(bytes, bytesRead);
+			}
+
+			return bytes;
+		} catch (IOException e) {
+			logger.error("IOException occurred while reading sample buffer", e);
+			return new byte[0];
+		} finally {
+			FileHelper.safeClose(fileInputStream);
```

Comment:
You should be able to skip the "safeClose" and use try-with-resources directly.

In fact, there's nothing safe about safeClose in my mind. It swallows exceptions and logs them at debug level.